editor/src/components/navigator/navigator-item/navigator-item.tsx

```diff
@@ -140,7 +140,7 @@ function selectItem(
   index: number,
   selected: boolean,
   event: React.MouseEvent<HTMLDivElement>,
-  conditionalOverrideUpdate: ConditionalCase | 'clear-override' | 'no-update',
+  conditionalOverrideUpdate: ConditionalOverrideUpdate,
```

Comment:
this functionality should be moved to a dedicated click handler in `ConditionalClauseNavigatorItemContainer`

Reply:
https://github.com/concrete-utopia/utopia/pull/3740/commits/b548e36b8595134b86e87f91789fa16e557c7b39

Reply:
Note @bkrmendy : I reverted the refactor you requested (https://github.com/concrete-utopia/utopia/pull/3740/commits/930dbee776c53dae7893b6759cba93c07ed73435) because it would still miss the handling of clicking the individual branch items. I still think you're right and the logic should live in the `ConditionalsClauseNavigatorItemContainer` but it's a bigger refactor and I think it should live in its own PR.

Reply:
Sounds good